File: 프로그래머스/2/131127. 할인 행사/할인 행사.java
import java.util.*;
class Solution {
    public int solution(String[] want, int[] number, String[] discount) {
        HashMap<String,Integer> map = new HashMap<>();
        
        int answer = 0;
        for(int i = 0; i < want.length; i++) {
            map.put(want[i],number[i]);
        }
        for(int i = 0; i <= discount.length-10; i++) {
            HashMap<String,Integer> disMap = new HashMap<>();
            for(int j = i; j < i + 10; j++) {
                if(!map.containsKey(discount[j])) break;
                disMap.put(discount[j], disMap.getOrDefault(discount[j],0) + 1);
            }
            boolean check = true;
            for(String key : map.keySet()) {
                if(map.get(key) != disMap.get(key)) {
                    check = false;
                    break;
                }
            }
            if(check) {
                answer++;
                
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function isEqual(obj1, obj2) {
    if (Object.keys(obj1).length !== Object.keys(obj2).length)
        return false;
    
    for (const key in obj1) {
        if (obj1[key] !== obj2[key]) return false;
    }
    return true;
}

function solution(want, number, discount) {
    var answer = 0;
    const obj1 = {};
    for (let i = 0; i < want.length; i++) {
        obj1[want[i]] = number[i];
    }
    
    for (let i = 0; i <= discount.length - 10 ;i++) {
        const obj2 = {};
        for (let j = i; j < i + 10; j++) {
            obj2[discount[j]] = (obj2[discount[j]] || 0) + 1;
        }
        if (isEqual(obj1,obj2)) answer++;
    }
    return answer;
}